rdd/rdd.py

```diff
@@ -180,3 +180,81 @@ def bin_data(data, yname, xname, bins=50, agg_fn=np.mean):
             binned_df.loc[binned_df.index[i], xname] = bin_midpoint[i]
             binned_df.loc[binned_df.index[i], 'n_obs'] = dat_temp.shape[0]
     return binned_df
+
+
+def visual_balance_check(input_data, xname, zname, cut=0):
+    '''
+    This function creates and displays a plot for visual inspection of balance for a control 
+    variable of interest by simply plotting the conditional mean of the control variable with 
+    respect to the running variable and the treatment condition.
+
+    INPUT:
+        input_data: dataset with outcome, running variables, and controls (pandas DataFrame)
+        xname: name of running variable (string)
+        zname: name of control variable (string)
+        cut: location of threshold in xname (scalar) (default is 0)
+
+    OUTPUT:
+        No object or value is returned. Only a plot is displayed
+    '''
+
+    data = input_data.copy() # To avoid SettingWithCopy warnings
+    #Treated group
+    trt = input_data[input_data['X'] >= cut]
+    #Control group
+    control = input_data[input_data['X'] < cut]
+
+    '''Get conditional mean for covariate of interested with respect to
+        value of running variable and treatment condition. 
+    '''
+    trt_conditional_mean = trt.groupby('X').mean()
```

Comment:
@divyansha I'm worried how this will play out when X is continuous -- it could be very noisy, and in many cases it would simply be the mean of 1 observation.  Could we instead cut the data into 100/500/etc bins?

Reply:
That sounds good to me - maybe we could do a default of 30-50